### PR TITLE
Added all functionallity needed for AdService

### PR DESCRIPTION
**Pull request by Danryg**
Created so that we can easily ad, remove and edit ads in our firestore

### CODE REVIEW

File: app/src/services/AdService.ts [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
 
+/**
+ * Handles all fetching and publishing of ads
+ *
+ * @version 1.0.0
+ * @author [Daniel Rygaard](https://github.com/Danryg)
+ * @author [Johan Blickhammar](https://github.com/JohanBlickhammar)
+ */
 export default class AdService {
-  static async addAd(ad: Ad): Promise<ServiceSuccessResponse> {
```
Comment (emilsvennesson) [inline app/src/services/AdService.ts:20]: I don't think you should need to specify and UID to publish an ad

File: app/src/services/AdService.ts [hunk 2]
```diff
@@ -30,4 +45,138 @@ export default class AdService {
...
     }
   }
+
+  /**
+   * This methods get all the ads from the connected firebase
+   * @param book : Book (optional) will find ads with this book
+   * @param user : User (optional) will find ads with this user
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  private static async getAds(
```
Comment (emilsvennesson) [inline app/src/services/AdService.ts:55]: maybe move some logic to getAdsFromBook and getAdsFromUser

File: app/src/services/AdService.ts [hunk 3]
```diff
@@ -30,4 +45,138 @@ export default class AdService {
...
+    return { success: true, data: ads };
+  }
+
+  /**
+   * This edits a document in the connected Firebase Firestore
+   * @param ad the ad that should be edited
+   * @param data the data that should be changed, Ex input {condition: newCondition} or {price: newPrice}
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  private static async editAd(
```
Comment (emilsvennesson) [inline app/src/services/AdService.ts:90]: I think the UID should be enough to edit an ad, it doesn't make sense that you would need an entire ad object

File: app/src/services/AdService.ts [hunk 4]
```diff
@@ -30,4 +45,138 @@ export default class AdService {
...
+  static async getAdsFromBook(book: Book): Promise<ServiceSuccessResponse> {
+    return this.getAds(book);
+  }
+
+  /**
+   * Removes an add from the connected Firebase
+   * @param ad the ad that will be deleted
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  static async removeAd(ad: Ad): Promise<ServiceSuccessResponse> {
```
Comment (emilsvennesson) [inline app/src/services/AdService.ts:136]: same here, UID

File: app/src/services/AdService.ts [hunk 5]
```diff
@@ -30,4 +45,138 @@ export default class AdService {
...
+    return success;
+  }
+
+  /**
+   * Edits the price of an ad
+   * @param ad the ad that will be altered
+   * @param newPrice the new price that will be set on the ad
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  static async editAdPrice(
```
Comment (emilsvennesson) [inline app/src/services/AdService.ts:150]: UID

File: app/src/services/AdService.ts [hunk 6]
```diff
@@ -1,23 +1,38 @@
...
-  tags: string[];
-  publishDate: Date;
-  wear: number;
-  isbn: number;
+  uid: string;
+  userId: string;
+  bookId: string;
+  price: number;
+  condition: string;
+  conditionDescribtion: string;
```
Comment (EliasFalk) [inline app/src/services/AdService.ts:24]: conditionDescribtion -> conditionDescription

File: app/src/services/AdService.ts [hunk 7]
```diff
@@ -30,4 +45,138 @@ export default class AdService {
...
+    return this.editAd(ad, { condition: newCondition });
+  }
+
+  /**
+   * Edits the condition describtion of an ad
+   * @param ad the ad that will be altered
+   * @param newConditionDescribtion the new condition describtion that will be set on the ad
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  static async editAdConditionDescribtion(
```
Comment (emilsvennesson) [inline app/src/services/AdService.ts:176]: spelling mistake, Description

UID

File: app/src/services/ServiceSuccessResponse.ts [hunk 8]
```diff
@@ -1,4 +1,5 @@
 export default interface ServiceSuccessResponse {
   success: boolean;
+  data?: any;
```
Comment (emilsvennesson) [inline app/src/services/ServiceSuccessResponse.ts:3]: can this be more specific?

**Review by emilsvennesson — CHANGES_REQUESTED** [review]

File: app/src/services/AdService.ts (hunk 3 above)
Comment (Danryg) [reply app/src/services/AdService.ts:90]: Fixed

File: app/src/services/AdService.ts (hunk 4 above)
Comment (Danryg) [reply app/src/services/AdService.ts:136]: Fixed

File: app/src/services/AdService.ts (hunk 7 above)
Comment (Danryg) [reply app/src/services/AdService.ts:176]: fixed

File: app/src/services/AdService.ts (hunk 5 above)
Comment (Danryg) [reply app/src/services/AdService.ts:150]: fixed

File: app/src/services/ServiceSuccessResponse.ts (hunk 8 above)
Comment (Danryg) [reply app/src/services/ServiceSuccessResponse.ts:3]: fixed

File: app/src/services/AdService.ts (hunk 1 above)
Comment (EliasFalk) [reply app/src/services/AdService.ts:20]: Agreed

**Pull request merged**